aspnetcore/blazor/common/samples/3.x/BlazorServerEFCoreSample/BlazorServerDbContextExample/Pages/ViewContact.razor

```diff
@@ -0,0 +1,196 @@
+﻿@page "/View/{ContactId:int}"
+
+@using Microsoft.EntityFrameworkCore
+
+@inject IDbContextFactory<ContactContext> DbFactory
+@inject IPageHelper PageHelper
+@inject EditSuccess EditSuccessState
+
+@if (NotFound && !Loading)
+{
+    <div class="alert alert-warning">A contact with that id doesn't exist.</div>
+}
+else
+{
+    @if (Contact == null || Loading)
+    {
+        <div class="alert alert-info">Loading...</div>
+    }
+    else
+    {
+        <h1>@Contact.LastName, @Contact.FirstName</h1>
+        <br />
+        @if (EditSuccess)
+        {
+            <div class="alert alert-success">Successfully updated the contact.</div>
+        }
+        <hr />
+        <h2>📞 Phone</h2>
+        <br />
+        <h3>@Contact.Phone</h3>
+        <br />
+        <h2>🏠 Address</h2>
+        <br />
+        <h3>@Contact.Street</h3>
+        <h3>@Contact.City, @Contact.State @Contact.ZipCode</h3>
+        <br />
+        <a class="btn btn-primary" title="Go Back" href="/@PageHelper.Page">
+            🔙 Back to Grid
+        </a>
+        @if (!Deleted)
+        {
+            <a class="btn btn-primary" href="/edit/@Contact.Id" title="Edit">Edit</a><span>&nbsp;</span>
+            @if (!ShowConfirmation)
+            {
+                <button class="btn btn-danger" @onclick="@(() => { ShowConfirmation = true; EditSuccess = false; })">Delete</button>
+            }
+            else
+            {
+                <br />
+                <br />
+                <DeletePrompt Confirmation="@(async result => await ConfirmAsync(result))" />
+            }
+        }
+        else
+        {
+            <div class="btn btn-danger">🚫 This item has been deleted.</div>
+        }
+    }
+}
+
+@code {
+    /// <summary>
+    /// Tracking of asynchronous calls.
+    /// </summary>
+    public bool Loading;
+
+    /// <summary>
+    /// Id from route of contact.
+    /// </summary>
+    [Parameter]
+    public int ContactId { get; set; }
+
+    /// <summary>
+    /// Navigation check.
+    /// </summary>
+    private int _lastContactId;
+
+    /// <summary>
+    /// True with redirect from successful edit.
+    /// </summary>
+    [Parameter]
+    public bool EditSuccess { get; set; }
+
+    /// <summary>
+    /// Contact entity.
+    /// </summary>
+    public Contact Contact { get; set; }
+
+    /// <summary>
+    /// Set to true when not found.
+    /// </summary>
+    public bool NotFound = false;
+
+    /// <summary>
+    /// Set to true when delete is requested.
+    /// </summary>
+    public bool ShowConfirmation = false;
+
+    /// <summary>
+    /// Set to true when delete is successful
+    /// </summary>
+    public bool Deleted = false;
```

Comment:
This looks like the state on the component, we don't recommend making this public

Reply:
Got it ... will review all components and update as needed.

Reply:
@javiercn it is state but it manipulates the UI, wouldn't it make sense to keep public for testing? Otherwise should I make methods private too?

Reply:
@JeremyLikness If its public, it can be manipulated from outside the component, which will has the potential to break the component rendering model in the sense that you might expect a change in the property to trigger a re-render. We normally discourage such behaviors since a component should be responsible to decide when it needs to render itself.

Exposing something like that gives consumers the impression that they can take a `@ref` to the component and modify the state to cause a change, which is not the case.

The general recommendation is that components are explicit about what API they expose to other components.

Reply:
Addressed in latest commit.